src/macros.jl

```diff
@@ -2748,3 +2748,215 @@ julia> @rdistinct!(df, :x + :y)
+    quote            
+        $DataFrames.rename($x, $(t...))
+    end    
+end
+
+"""
+    @rename(d, args...)
+
+Change column names.
+
+### Arguments
+
+* `d` : an AbstractDataFrame
+* `args...` : expressions of the form `:new = :old` specifying the change of a column's name 
+from "old" to "new". The left- and right-hand side of each expression can be passed as
+symbol arguments, as in `:old_col`, or strings escaped with `$DOLLAR` as in `$DOLLAR"new_col"`.
+See  **Details** for a description of accepted values.
+
+### Returns
+
+* `::AbstractDataFrame`
+
+Inputs to `@rename` can come in two formats: a `begin ... end` block, or as a series of 
+keyword-like arguments. For example, the following are equivalent:
+
+```julia
+@rename df begin 
+    :new_col = :old_col
+end
+```
+
+and 
+
+```
+@rename(df, :new_col = :old_col)
+```
+
+### Details
+
+Both the left- and right-hand side of an expression specifying a column name assignment 
+can be either a `Symbol` or a `String`` escaped with `$DOLLAR` For example `:new = ...`, 
+and `$(DOLLAR)"new" = ...` are both valid ways of assigning a new column name.
+
+This idea can be extended to pass arbitrary right-hand side expressions. For example, 
+the following are equivalent:
+
+```
+@rename(df, :new = :old1)
+```
+
+and
+
+```
+@rename(df, :new = $("old_col" * "1"))
+```
+
+### Examples
+```
+julia> df = DataFrame(old_col1 = rand(5), old_col2 = rand(5),old_col3 = rand(5));
+
+julia> @rename(df, :new1 = :old_col1)
+5×3 DataFrame
+ Row │ new1       old_col2   old_col3 
+     │ Float64    Float64    Float64  
+─────┼────────────────────────────────
+   1 │ 0.0176206  0.493592   0.348072
+   2 │ 0.861545   0.512254   0.85763
+   3 │ 0.263082   0.0267507  0.696494
+   4 │ 0.643179   0.299391   0.780125
+   5 │ 0.731267   0.18905    0.767292
+
+julia> @rename(df, :new1 = :old_col1, :new2 = $DOLLAR"old_col2")
+5×3 DataFrame
+ Row │ new1       new2       old_col3 
+     │ Float64    Float64    Float64  
+─────┼────────────────────────────────
+   1 │ 0.0176206  0.493592   0.348072
+   2 │ 0.861545   0.512254   0.85763
+   3 │ 0.263082   0.0267507  0.696494
+   4 │ 0.643179   0.299391   0.780125
+   5 │ 0.731267   0.18905    0.767292
+
+julia> @rename(df, :new1 = $DOLLAR("old_col" * "1"), :new2 = :old_col2)
+5×3 DataFrame
+ Row │ new1       new2       old_col3 
+     │ Float64    Float64    Float64  
+─────┼────────────────────────────────
+   1 │ 0.0176206  0.493592   0.348072
+   2 │ 0.861545   0.512254   0.85763
+   3 │ 0.263082   0.0267507  0.696494
+   4 │ 0.643179   0.299391   0.780125
+   5 │ 0.731267   0.18905    0.767292
+```
+"""
+macro rename(x, args...)
+    esc(rename_helper(x, args...))        
+end
+
+function rename!_helper(x, args...)
+    x, exprs, outer_flags, kw = get_df_args_kwargs(x, args...; wrap_byrow = false)    
+    t = (fun_to_pair(ex) for ex in exprs)            
+    quote            
+        $DataFrames.rename!($x, $(t...))
```

Comment:
Extended from comments below. Instead of doing `String(lhs) => String(rhs)` in the parsing stage, let's clean up the inputs all together. 

I would allocate an intermediate array of renaming calls. then write a function that turns every vector of pairs into a vector of pairs of strings. Then pass that to `rename`. Something like

```
julia> function pairs_to_str_pairs(args...)
           map(args) do arg
               if !(arg isa Pair)
                   throw(ArgumentError("Non-pair created in @rename"))
               end
                   
               string(first(arg)) => string(last(arg))
           end
       end;
```

then do 

```
        $DataFrames.rename!($x, $pairs_to_str_pairs($(t...)...))
```

This means you don't have to do `MacroTools.postwalk` in the parsing stage. See the function `make_source_concrete` for inspiration. It's used when we have more than one input column, i.e. `@rtransform df :x = :a + :b`

Edge case: What happens when we want to rename the first column, i.e. `@rename df :a = $1`. I want this to work, particularly because `rename(df, 1 => :a)` works. But I don't want to mess around with `names(df, 1)` in this stage. So its a tough balance. Fortunately we get to point fingers at DataFrames.jl for any limitations.

Reply:
I've resolved your comments aside from this edge case. Could we handle this case with a modified version 
of `pairs_to_str_pairs`? 

```
function pairs_to_str_pairs(args...)
           map(args) do arg
               if !(arg isa Pair)
                   throw(ArgumentError("Non-pair created in @rename"))
               end              
               if first(arg) isa Int            
                   return first(arg) => string(last(arg))
               end      
               string(first(arg)) => string(last(arg))
           end
end
```

Reply:
Yes this seems good!

Remember, this is done at call-time, not at the parsing stage.